### ISSUE TITLE
UserAgent and other headers ignored after first redirect

### ISSUE THREAD

**Issue by farwayer** [levigross/grequests#12]
Look like problem in `doFollowingRedirects()`

```go
...
		if redirect != 0 {
			nreq := new(Request)
			nreq.Method = ireq.Method
			if ireq.Method == "POST" || ireq.Method == "PUT" {
				nreq.Method = "GET"
			}
			nreq.Header = make(Header)
			nreq.URL, err = base.Parse(urlStr)
			if err != nil {
				break
			}
			if len(via) > 0 {
				// Add the Referer header.
				lastReq := via[len(via)-1]
				if ref := refererForURL(lastReq.URL, nreq.URL); ref != "" {
					nreq.Header.Set("Referer", ref)
				}
...
```
There is only `Referer` header is set.

**Comment by farwayer** [comment]
Oh, sorry. Look like it is upstream issue.

**Comment by levigross** [comment]
Hello!

Thanks for pointing this out, I will look into this and (hopefully) push a fix within a few days

Levi

**Comment by levigross** [comment]
This issue has been fixed in 0.6
